tests/unit/serve/instrumentation/test_gateway_metric_labels.py

```diff
@@ -100,15 +100,56 @@ def meow(self, docs, **kwargs):
             ]
         )
 
-        assert (
-            gateway_metric_data_point['jina_received_response_bytes'][0]['attributes'][
-                'deployment'
-            ]
-            == 'first_exec'
-        )
-        assert (
-            gateway_metric_data_point['jina_received_response_bytes'][1]['attributes'][
-                'deployment'
-            ]
-            == 'second_exec'
+        assert {'first_exec', 'second_exec'} == {
+            i['attributes']['deployment']
+            for i in gateway_metric_data_point['jina_received_response_bytes']
+        }
+        assert {'first_exec', 'second_exec'} == {
+            i['attributes']['deployment']
+            for i in gateway_metric_data_point['jina_sent_request_bytes']
+        }
+        assert {'first_exec', 'second_exec'} == {
+            i['attributes']['deployment']
+            for i in gateway_metric_data_point['jina_sending_request_seconds']
+        }
+
+
+def test_merge_with_no_reduce(monkeypatch_metric_exporter):
+    collect_metrics, read_metrics = monkeypatch_metric_exporter
+
+    f = (
+        Flow(
+            tracing=False,
+            metrics=True,
+            metrics_exporter_host='localhost',
+            metrics_exporter_port=4317,
+            port=12345,
         )
+        .add(name='name1')
+        .add(name='name2', needs=['gateway'])
+        .add(name='name3', needs=['name1', 'name2'], disable_reduce=True)
+    )
+    with f:
+        f.post('/')
+        collect_metrics()
+        metrics = read_metrics()
+
+        gateway_metrics = metrics['gateway/rep-0'][0]['resource_metrics'][0][
+            'scope_metrics'
+        ][0]['metrics']
+        gateway_metric_data_point = {
+            i['name']: i['data']['data_points'] for i in gateway_metrics
+        }
+
+        assert {'name1', 'name2', 'name3'} == {
```

Comment:
can u do these assertions outside the context manager? I believe, if there is an error and ctxt manager is left, it may not be called